lib/src/Uncertainty/Model/DistributionImplementation.cxx

```diff
@@ -648,6 +648,18 @@ Sample DistributionImplementation::getSampleByQMC(const UnsignedInteger size) co
   return returnSample;
 }
 
+Function DistributionImplementation::getPDF() const {
+  return Function(DistributionImplementation::PDFWrapper(*this));
+}
+
+Function DistributionImplementation::getLogPDF() const {
+  return Function(DistributionImplementation::LogPDFWrapper(*this));
+}
+
+Function DistributionImplementation::getCDF() const {
+  return Function(DistributionImplementation::CDFWrapper(*this));
+}
+
 /* Get the DDF of the distribution */
```

Comment:
static const Factory<PDFWrapper> Factory_PDFWrapper;

should be added at the end of this file

Reply:
@jschueller Where should this line be added? When I add it to DistributionImplementation.cxx I get the following error upon compilation:

```
DistributionImplementation.cxx:88:22: error: ‘PDFWrapper’ was not declared in this scope
   88 | static const Factory<PDFWrapper> Factory_PDFWrapper;
```

Reply:
maybe try with `Factory<DistributionImplementation::PDFWrapper>`

Reply:
This seems to work, however the PDFWrapper::load method is not called. My guess is that the getClassName method returns "EvaluationImplementation" rather than "PDFWrapper"

Reply:
indeed, did you try to overload it ?

Reply:
it can be that the GetClassName static method is needed, you might need to use CLASSNAME / CLASSNAMEINIT macros

Reply:
I did, here is the result:

```
InvalidArgumentException : No object with label 'instance' in study
```

Reply:
try CLASSNAME / CLASSNAMEINIT

Reply:
OK, this works, but now I get segmentation faults because the loaded object is a DistributionImplementation, not an instance of a child class (e.g. Normal, Uniform...): thus computePDF does not work.

Reply:
I am starting to think this is not worth it: we could simply have the save() method throw.

Reply:
it should be a Distribution

Reply:
Actually, my first try was simply this;

```
    /* Method save() stores the object through the StorageManager */
    void save(Advocate & adv) const override
    {
      EvaluationImplementation::save(adv);
      adv.saveAttribute("p_shared_distribution_", p_shared_distribution_);
    }

    /* Method load() reloads the object from the StorageManager */
    void load(Advocate & adv) override
    {
      EvaluationImplementation::load(adv);
      adv.loadAttribute("p_shared_distribution_", p_shared_distribution_); // the compiler dislikes this line
    }
```

but this causes an error message during compilation:

```
StorageManager.hxx: In instantiation of ‘OT::Advocate& OT::Advocate::loadAttribute(const String&, _Tp&) [with _Tp = OT::Pointer<OT::DistributionImplementation>; OT::String = std::__cxx11::basic_string<char>]’:
openturns/DistributionImplementation.hxx:988:73:   required from here
openturns/lib/src/Base/Common/openturns/StorageManager.hxx:371:23: error: cannot bind non-const lvalue reference of type ‘OT::Bool&’ {aka ‘bool&’} to an rvalue of type ‘bool’
  371 |     mgr_.readAttribute( p_state_, attributeName, attributeValue );
```

Reply:
yes, you cannot save a pointer directly

Reply:
> it should be a Distribution

But we are in DistributionImplementation.hxx, Distribution is not actually defined here:

```
 error: variable ‘OT::Distribution savedDistribution’ has initializer but incomplete type
  975 |       Distribution savedDistribution(p_shared_distribution_);
```

Reply:
maybe you can try with a `TypedInterfaceObject<DistributionImplementation>`

Reply:
OK, it works that way. Thanks a lot Julien!